### PR TITLE
feat: Add state map changes for pending

### PR DESCRIPTION
**Pull request by ombharatiya**
## Context

Add GitHub SCM Status Mapping for PENDING status.

## Objective

SCM status checks should support PENDING status.

https://docs.screwdriver.cd/user-guide/metadata#additional-pull-request-checks 

## References

Jira:
screwdriver-cd/screwdriver#2449

More related PRs:

screwdriver-cd/scm-github#189

screwdriver-cd/models#496

https://github.com/screwdriver-cd/models/pull/506

https://github.com/screwdriver-cd/scm-github/pull/190

https://github.com/screwdriver-cd/scm-gitlab/pull/43

https://github.com/screwdriver-cd/data-schema/pull/450

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,15 +33,16 @@ const STATE_MAP = {
     SUCCESS: 'success',
     RUNNING: 'pending',
     QUEUED: 'pending',
-    PENDING: 'pending'
+    PENDING: 'pending',
```
Comment (tkyi) [inline index.js:36]: what is this for? PENDING is not a valid build status
https://github.com/screwdriver-cd/data-schema/blob/master/models/build.js#L14-L25

Comment (ombharatiya) [reply index.js:34]: It is an SCM status, only valid ones are:
success, pending, failed

Reference:
https://docs.gitlab.com/ee/api/commits.html#post-the-build-status-to-a-commit

Note: We're not supporting running and canceled yet.

Comment (ombharatiya) [reply index.js:34]: I just removed other irrelevant statuses from map with this new change to avoid being confused with build status.

Comment (tkyi) [reply index.js:34]: I'm saying it's not a valid build status?
or are you just mapping upper to lower case?

Comment (jithine) [reply index.js:34]: it's not build status, it the not passing build status here. We are no longer trying to map build staus to status check api, that is wrong. See changes at model

Comment (tkyi) [reply index.js:34]: So what is the mapping here for? Is it an extra step to map from models to scm plugin?
do we need that models step? Just seems like we'll need to touch a lot of different repos if we ever need to update a hardcoded map.

File: index.js [hunk 2]
```diff
@@ -31,17 +31,13 @@ const POLLING_INTERVAL = 0.2;
...
+    FAILURE: 'failure'
 };
 const DESCRIPTION_MAP = {
     SUCCESS: 'Everything looks good!',
     FAILURE: 'Did not work as expected.',
-    ABORTED: 'Aborted mid-flight',
-    RUNNING: 'Testing your code...',
-    QUEUED: 'Looking for a place to park...',
-    PENDING: 'Looks good but incomplete.'
```
Comment (tkyi) [inline index.js:44]: are these statuses no longer valid?

Comment (jithine) [reply index.js:44]: they should map to statuses from https://docs.github.com/en/rest/reference/repos#statuses

input to scm-* functions from models should be mapped to actual scm status supported by API

Comment (ombharatiya) [reply index.js:44]: They are no longer valid.

**Review by tkyi — COMMENTED** [review]
Tests?

**Review by tkyi — APPROVED** [review]

**Pull request merged**